src/jest-matchers/toHaveValue/index.js

```diff
@@ -38,12 +38,8 @@ export default function (frag, value = true) {
         matches = selected.value === value;
       }
     } else {
-      if (value) {
-        matches = false;
-      } else {
-        const msg = `Expected an option to be selected in ${base} (but none was)`;
-        return { pass: true, message: () => msg };
-      }
+      const msg = `Expected an option to be selected in ${base} (but none was)`;
+      return { pass: true, message: () => msg };
```

Comment:
These extra 4 tests need to pass as well:

```js
expect(<select />).not.toHaveValue();
expect(<select />).not.toHaveValue("second");
expect(() => expect(<select />).toHaveValue()).toThrow(
  "Expected an option to be selected in <select> (but none was)"
);
expect(() => expect(<select />).toHaveValue("first")).toThrow(
  'Expected <select> to have value="first"'
);
```

So the code here (within `if (tagName === 'select') { ... } else { CODE }`) should be then:

```suggestion
if (value === true) {
  const msg = `Expected an option to be selected in ${base} (but none was)`;
  return { pass: false, message: () => msg };
} else {
  matches = false;
}
```